app/controllers/admin/videos_controller.rb

```diff
@@ -0,0 +1,32 @@
+class Admin::VideosController < ApplicationController
+  before_filter :require_user
+  before_filter :require_admin
+
+  def new
+    @video = Video.new
+  end
+
+  def create
+    @video = Video.new(video_params)
+    if @video.save
+      flash[:success] = "You have successfully add the video #{@video.title}."
+      redirect_to new_admin_video_path
+    else
+      flash[:error] = "You cannot add this video, Please check errors."
```

Comment:
Make sure you use `flash.now` when using `render`. If you don't, the flash message will display on the next request as well.